commands/ls.go

```diff
@@ -213,7 +213,17 @@ type lsContext struct {
 }
 
 func (c *lsContext) MarshalJSON() ([]byte, error) {
-	return json.Marshal(c.Builder)
+	// can't marshal c.Builder directly because Builder type has custom MarshalJSON
+	dt, err := json.Marshal(c.Builder.Builder)
+	if err != nil {
+		return nil, err
+	}
+	var m map[string]any
+	if err := json.Unmarshal(dt, &m); err != nil {
+		return nil, err
+	}
+	m["Current"] = c.Builder.Current
+	return json.Marshal(m)
```

Comment:
Oh interesting, this happens when marshaling structs that embed pointers. What do you think of this:

```go
func (c *lsContext) MarshalJSON() ([]byte, error) {
	type Alias builder.Builder
	return json.Marshal(&struct {
		*Alias
		Current bool
	}{
		Alias:   (*Alias)(c.Builder.Builder),
		Current: c.Builder.Current,
	})
}
```

Reply:
This doesn't work though. If you the original `Builder.MarshalJSON` does not get called anymore. https://go.dev/play/p/eJgDvB1mVAh

Reply:
Yes indeed